### ISSUE TITLE
Cannot negate byte in array

### ISSUE THREAD

**Issue by markjreed** [irmen/prog8#84]
This code doesn't compile:

```
main {
  byte[1] foo
  sub start() {
    foo[0] = 1
    foo[0] = -foo[0]
  }
}
```

Result:
```* internal error *
Exception in thread "main" prog8.code.core.AssemblyError: no asm gen for in-place negate byte
        at prog8.codegen.cpu6502.assignment.AugmentableAssignmentAsmGen.inplaceNegate$codeGenCpu6502(AugmentableAssignmentAsmGen.kt:1901)```

**Comment by irmen** [comment]
More problematic assignments;

```
%import floats

main {
  byte[1] foo
  ubyte[1] foou
  word[1] foow
  uword[1] foowu
  float[1] flt
  byte d

  sub start() {
    foo[0] = -d         ; ... why is this generating an array in-place modification operation?
    foo[0] = -foo[0]
    foou[0] = ~foou[0]
    foow[0] = -foow[0]
    foowu[0] = ~foowu[0]
    flt[0] = -flt[0]            ; TODO also fix crash when selecting vm target: fpReg1 out of bounds
  }
}
```

**Comment by irmen** [comment]
float array problems moved to separate issue #85

**Comment by irmen** [comment]
fixed by 434515d9578e011a67735c44aa4c539f73774ea9 and 094f7803b7bcd9e0e015ca6fc0d9ef1c633c8e22 and b0c7bad3910745e637f8f8e41ddf1f7c1751e5d4
